prql-compiler/src/sql/translator.rs

```diff
@@ -389,16 +404,52 @@ fn prepend_with_from(pipeline: &mut Pipeline, table: Option<TableRef>) {
     }
 }
 
+/// Aggregate several ordered ranges into one, computing the intersection.
+///
+/// Returns a tuple of `(start, end)`, where `end` is optional.
+fn range_of_ranges(ranges: Vec<Range>) -> Result<(i64, Option<i64>)> {
```

Comment:
This seems like a very complicated implementation for what it's actually doing, but I'm not sure of an easier one...